prow/plugins/caicloud-cla/cla_test.go

```diff
@@ -0,0 +1,292 @@
+package caicloudcla
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/github"
+)
+
+type ghc struct {
+	*testing.T
+	labels        sets.String
+	commit        *github.CommitData
+	pr            *github.PullRequest
+	isMember      bool
+	LabelsAdded   []string
+	LabelsRemoved []string
+	createCommentErr, addLabelErr, removeLabelErr,
+	getIssueLabelsErr, getCommitErr, getPullRequestErr, isMemberErr error
+}
+
+func (c *ghc) CreateComment(_, _ string, _ int, _ string) error {
+	c.T.Logf("CreateComment")
+	return c.createCommentErr
+}
+
+func (c *ghc) AddLabel(_, _ string, _ int, label string) error {
+	c.T.Log("AddLabel")
+
+	c.LabelsAdded = append(c.LabelsAdded, fmt.Sprintf("%s", label))
+	return nil
+
+	return c.addLabelErr
+}
+
+func (c *ghc) RemoveLabel(_, _ string, _ int, label string) error {
+	c.T.Log("RemoveLabel")
+	c.LabelsRemoved = append(c.LabelsRemoved, fmt.Sprintf("%s", label))
+	return c.removeLabelErr
+}
+
+func (c *ghc) GetIssueLabels(_, _ string, _ int) (ls []github.Label, err error) {
+	c.T.Log("GetIssueLabels")
+	for label := range c.labels {
+		ls = append(ls, github.Label{Name: label})
+	}
+
+	err = c.getIssueLabelsErr
+	return
+}
+
+func (c *ghc) GetPullRequest(_, _ string, _ int) (*github.PullRequest, error) {
+	c.T.Log("GetPullRequestChanges")
+	return c.pr, c.getPullRequestErr
+}
+
+func (c *ghc) GetCommit(_, _, _ string) (*github.CommitData, error) {
+	c.T.Log("GetCommit")
+	return c.commit, c.getCommitErr
+}
+
+func (c *ghc) IsMember(_, _ string) (bool, error) {
+	c.T.Log("IsMember")
+	return c.isMember, c.isMemberErr
+}
+
+func TestHandlePR(t *testing.T) {
+	cases := []struct {
+		name          string
```

Comment:
Define the `cases` as a map, then move the name out of the struct, and make it as the key of the map to describe the test case.

```go
	cases := map[string]struct {
		labels        sets.String
		isMember      bool
		action        github.PullRequestEventAction
		commit        *github.CommitData
		pr            *github.PullRequest
		addedLabels   []string
		removedLabels []string
		err           error
		createCommentErr, addLabelErr, removeLabelErr, getIssueLabelsErr, getCommitErr,
		getPullRequestErr, isMemberErr error
	}{
		"no label, not member": {
			labels:   sets.NewString(),
			isMember: false,
			commit: &github.CommitData{
				Author: github.Author{
					Email: "bmj@gmail.com",
				},
				Committer: github.Committer{
					Email: "bmj@gmail.com",
				},
			},
			pr: &github.PullRequest{
				Number: 101,
			},
			action:        github.PullRequestActionOpened,
			addedLabels:   []string{fmt.Sprintf("%s", claYesLabel)},
			removedLabels: nil,
		},
	}
````